### DISCUSSION TITLE
rke2-cilium cluster mesh possible?

### DISCUSSION THREAD

**Discussion by seimic** [rancher/rke2#1474] [Q&A] [answered]
Hi,

I'm not sure where to ask the question, so I do here.
Is it possible to configure a cluster mesh using the rke2-cilium chart? The chart bundled with rke2 seems to be different than the official cilium helm chart. At least I don't see any values in it to set up 'cilium managed etcd' which is one of the requirements to configure a cilium cluster mesh.

RKE2 cilium chart for version 1..9.8
https://github.com/rancher/rke2-charts/blob/main-source/packages/rke2-cilium/charts/values.yaml
and the official one for version 1.9.8
https://github.com/cilium/cilium/blob/v1.9.8/install/kubernetes/cilium/values.yaml

I try to configure the 'shared services' use case here: https://cilium.io/blog/2019/03/12/clustermesh#use-case-shared-services
Any suggestion would be appreciated.

Kind regards,
Michael

**Comment by vadorovsky** [comment, marked as answer]
Hi @seimic,

We intentionally wrap the official helm chart into our own one, which has less options, because we want to make sure that we expose only options which we (as SUSE/Rancher) are testing and supporting.

That said, I can work on adding cluster mesh support for next versions of rke2. But until it's done, you will have to use the upstream chart and skip the CNI deployment by setting the following in your `/etc/rancher/rke2/config.yaml` (on the node with rke2-server):

```yaml
cni: none
```

**Reply by KhalilSantana** [reply]
Hi, is this still the case as of RKE2 1.31.x?

**Reply by manuelbuil** [reply]
Now you have all options in the chart

**Comment by seimic** [comment]
Ahh ok, thanks. I give it a try with the upstream helm chart.
